### ISSUE TITLE
Add a button for a copy of the current page.

### ISSUE THREAD

**Issue by bruening-bw** [OpenBoard-org/OpenBoard#1181]
![feature request](https://github.com/user-attachments/assets/fbc1cafd-bce0-42ad-9d7e-894932f20fa3)

Instead of adding a new page I always make a copy of the current page. Then, instead of clearing the whole page, i clear only parts of it and leave the elements last drawn on the board for those students who are still busy copying. This is helpful for students and I don't have to wait for everybody to finish copying. And, by the way, it is closer to the use of a chalkboard.

The button would speed up my workflow a little bit. Would creating the button a big issue?

**Comment by letsfindaway** [comment]
There already is this possibility!

Just make a long press on the "Add Page" button and you get a popup menu. The middle option duplicates the current page and makes the duplicate the current page.
![image](https://github.com/user-attachments/assets/b5e5b146-1447-4a88-968a-de6e609a93c7)

**Comment by bruening-bw** [comment]
Ah, I didn't know that. But still it would be cool to have the button in the taskbar to make it faster. There would be enough space in the bar for users with FullHD devices. But I understand that it is not cool for lower resolutions. Thanks anyway.

**Comment by letsfindaway** [comment]
Yes, there are already low-res users complaining about too many buttons. But I have another idea.

You can also assign a keyboard shortcut to that action:

- [Locate](https://github.com/OpenBoard-org/OpenBoard/wiki/Configuration) your `OpenBoardUser.config` and add the following entry:
```ini
[Shortcut]
actionDuplicatePage=Alt+D, 0, 0
```
- Then pressing `Alt+D` will duplicate the current page

This works starting with OpenBoard 1.7.2.

**Comment by bruening-bw** [comment]
Cool! A shortcut is even better. I did a few minutes of research, and cool again! I didn't know about the many undocumented actions shortcuts can be attached to.

I am thinking about helping with the documentation. The Tutorial is not quite up to date. And since I have no coding skills it is the only thing I could contribute.

**Comment by bruening-bw** [comment]
I added the shortcut. But it is not working properly. It does the job when the left panel is open. Otherwise I have to press the shortcut twice to be effective.

What does ", 0, 0" do?

**Comment by letsfindaway** [comment]
> I added the shortcut. But it is not working properly. It does the job when the left panel is open. Otherwise I have to press the shortcut twice to be effective.

Hmm, for me it works at the first press of the shortcut. I don't know what I'm doing different.

> 
> What does ", 0, 0" do?

There are plans to add a User Interface to the shortcut configuration, but this needs more discussions how this should look like. Then it would not necessary to edit the settings file.

The `, 0, 0` are meant to assign Mouse or Stylus pen buttons to an action. But -- despite the zero -- it is not easy to describe what number you have to put there to assign one of the special keys of a mouse. So I cannot give a good description for that. In the end in  the GUI, you probably just press that button and OpenBoard inserts the proper number for you. In the meantime, just add `, 0, 0`. It is necessary to create the proper syntax.

**Comment by letsfindaway** [comment]
> Cool! A shortcut is even better. I did a few minutes of research, and cool again! I didn't know about the many undocumented actions shortcuts can be attached to.

The current list of supported actions can be seen here in the code:
https://github.com/OpenBoard-org/OpenBoard/blob/80fb7dd6e52bf2a2e2cf236f77f6628c0c12ec14/src/core/UBShortcutManager.cpp#L118-L204

I think for most of them you can at least imagine what they do. There are some hardcoded shortcuts too, which are mentioned in the code after that. Those shortcuts cannot be changed. 

> I am thinking about helping with the documentation. The Tutorial is not quite up to date. And since I have no coding skills it is the only thing I could contribute.

Documentation would always help!
